src/Core/Utils/BijectiveAssociation.hpp

```diff
@@ -0,0 +1,136 @@
+#pragma once
+
+#include <initializer_list>
+#include <map>
+
+namespace Ra {
+namespace Core {
+namespace Utils {
+
+/**
+ * \brief Bijective association on a finite set of <key, value> pairs.
+ * \tparam T1 Type of one element of the pair
+ * \tparam T2 Type of the other element of the pair
+ * \note If T1 and T2 refer to the same type, some operators will be removed.
+ */
+template <typename T1, typename T2, bool TypesAreEquals = std::is_same_v<T1, T2>>
+class BijectiveAssociation
+{
+  public:
+    using Type1 = T1;
+    using Type2 = T2;
+
+    /**
+     * \brief Constructor from { <T1, T2> } pairs
+     */
+    explicit BijectiveAssociation( std::initializer_list<std::pair<T1, T2>> pairs );
+    /**
+     * \brief Constructor from { <T2, T1> } pairs
+     * \note If T1 and T2 are the same type, this constructor is disabled.
+     */
+    explicit BijectiveAssociation( std::initializer_list<std::pair<T2, T1>> pairs );
```

Comment:
I think it might be confusing to forget about the direction key - value.
Having only one ctor is sufficient.
This comment is valid for other methods as well.

Reply:
I'll remove all overload and partial specialisation. I found some of them useful from a user point of vue (Given a unordered pair, if a have one element of this pair, I can get the other without inquiring which one is the key and which one is the value). But I agree to consider this as a function.